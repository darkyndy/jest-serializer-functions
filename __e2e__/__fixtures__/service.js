const service = {};

service.myArrowFunction = () => ('my-arrow-function-result');
service.myAnonymousFunction = function () {return 'my-anonymous-function-result'};
service.myNamedFunction = function named() {return 'my-named-function-result'};

export default service;
