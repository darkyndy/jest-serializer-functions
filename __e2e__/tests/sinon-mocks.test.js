import sinon from 'sinon';

// tested
import fixtures from './../__fixtures__/index';

// dependencies
import service from './../__fixtures__/service';

describe('fixtures - sinon mock', () => {
  let testSpecificMocks;

  beforeEach(() => {
    sinon.spy(service, 'myArrowFunction');
    sinon.spy(service, 'myAnonymousFunction');
    sinon.stub(service, 'myNamedFunction').returns('sinon-stub-result');

    testSpecificMocks = {};
  });
  afterEach(() => {
    service.myArrowFunction.restore();
    service.myAnonymousFunction.restore();
    service.myNamedFunction.restore();
  });

  it('static - function used as prop', () => {
    expect(fixtures.awesomeObj).toMatchSnapshot();
  });

  it('dynamic - function used as prop', () => {
    testSpecificMocks.obj = {
      fn: function x() {/*impl*/},
      spy: sinon.spy(),
      stub: service.myNamedFunction,
    };

    expect(fixtures.awesomeFn(testSpecificMocks.obj)).toMatchSnapshot();
  });
});

