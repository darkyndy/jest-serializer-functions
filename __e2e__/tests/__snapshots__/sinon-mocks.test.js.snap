// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures - sinon mock dynamic - function used as prop 1`] = `
Object {
  "fn": [Function] x,
  "spy": [Function],
  "stub": [Function] myNamedFunction,
}
`;

exports[`fixtures - sinon mock static - function used as prop 1`] = `
Object {
  "anonymous": [Function],
  "arrow": [Function],
  "named": [Function] named,
}
`;
